core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java

```diff
@@ -319,6 +323,32 @@ public Response getWebUIOverview() {
       mMetaMaster.getJournalSpaceMonitor().map(monitor ->
           response.setJournalDiskWarnings(monitor.getJournalDiskWarnings()));
 
+      Gauge entriesSinceGauge = MetricsSystem.METRIC_REGISTRY.getGauges()
+              .get(MetricKey.MASTER_JOURNAL_ENTRIES_SINCE_CHECKPOINT.getName());
+      Gauge lastCkPtGauge = MetricsSystem.METRIC_REGISTRY.getGauges()
+          .get(MetricKey.MASTER_JOURNAL_LAST_CHECKPOINT_TIME.getName());
+
+      if (entriesSinceGauge != null && lastCkPtGauge != null) {
+        long entriesSinceCkpt = (Long) entriesSinceGauge.getValue();
+        long lastCkptTime = (Long) lastCkPtGauge.getValue();
+        long timeSinceCkpt = System.currentTimeMillis() - lastCkptTime;
+        boolean over72Hrs = timeSinceCkpt > ServerConfiguration.getMs(
+            PropertyKey.MASTER_WEB_JOURNAL_CHECKPOINT_WARNING_THRESHOLD_TIME);
+        boolean passedThreshold = entriesSinceCkpt > ServerConfiguration
+            .getLong(PropertyKey.MASTER_JOURNAL_CHECKPOINT_PERIOD_ENTRIES);
+        if (lastCkptTime > 0 && passedThreshold && over72Hrs) {
```

Comment:
why lastCkptTime > 0 ?
How about the very first checkpoint?

Reply:
the `lastCkptTime` is an epoch timestamp. Basically 0 represents the Jan 1st 1970 date.

The metric is not updated when rebooting a master. The gauge is 0 until the first checkpoint is taken during the life of the process. Since we don't know when the last checkpoint was after a master is rebooted (it's not stored afaik?). I guess we can still issue the warning if a checkpoint hasn't been taken yet, I'll just need to adjust the error message to indicate no checkpoint taken since reboot. Maybe just with `N/A`?